src/main/java/io/supertokens/webserver/api/userroles/AddUserRoleAPI.java

```diff
@@ -59,9 +59,10 @@ protected void doPut(HttpServletRequest req, HttpServletResponse resp) throws IO
         }
 
         try {
-            UserRoles.addRoleToUser(main, userId, role);
+            boolean didUserHaveRole = !UserRoles.addRoleToUser(main, userId, role);
             JsonObject response = new JsonObject();
             response.addProperty("status", "OK");
+            response.addProperty("didUserHaveRole", didUserHaveRole);
```

Comment:
I think a better name would be `didUserAlreadyHaveRole`?

Reply:
agreed will make the change